src/api/system.c

```diff
@@ -1150,6 +1150,24 @@ static int f_text_input(lua_State* L) {
   return 0;
 }
 
+static int f_setenv(lua_State* L) {
+  size_t keylen, vallen;
+  const char *key = luaL_checklstring(L, 1, &keylen);
+  const char *val = luaL_checklstring(L, 2, &vallen);
+
+  int ok; // unused: this could be used later to report success
+#ifdef _WIN32
+  LPWSTR wkey = utfconv_utf8towc(key);
+  LPWSTR wval = utfconv_utf8towc(val);
```

Comment:
We should probably check the return values and report eventual failures.
I wonder if this should throw, instead of returning a failure value.